aliyun/src/main/java/org/apache/iceberg/aliyun/oss/OSSFileIO.java

```diff
@@ -89,7 +89,11 @@ public void deleteFile(String path) {
 
   private OSS client() {
     if (client == null) {
-      client = oss.get();
+      synchronized (this) {
+        if (client == null) {
+          client = oss.get();
+        }
+      }
```

Comment:
In the other PR, we were wondering whether we should just initialize the client eagerly in `initialize` What do you think about that vs. lazy initialization?

Reply:
As per my understanding the since s3 client is not serializable and initialize being called in only at driver when FileIO is initialized via CatlogUtil, we capture the s3 client against the wrapper `SerializableSupplier` and this will now be part of serialized FileIO obj (which will be used when it's broadcasted). Now in the executor the task threads starts calling the FileIO func directly, without calling intialize etc.

IIUC, it might not be possible to do it in initialize which is called at driver, Am I missing something here ?

Reply:
Ah, you're right. This is to handle serialization. Good catch.